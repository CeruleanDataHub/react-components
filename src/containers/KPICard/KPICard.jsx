import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";

import { Cell, Grid } from "../Grid/Grid";
import { Icon } from "../Icon/Icon";

const Container = styled.div`
  padding: 1.2em;
  font-family: Helvetica;
  font-size: 0.9rem;
`;

const colors = { red: "#ff0066", green: "#90ee7e", blue: "#1ea7fd" };

const Color = styled.span`
  color: ${({ textColor }) => textColor};
`;

const Value = styled.span`
  font-size: 2.2rem;
`;

const Bottom = styled.span`
  justify-content: flex-end;
  display: inline-flex;
  flex-direction: column;
`;

const FlexRow = styled.span`
  flex-direction: row;
`;

const TriangleIcon = styled.div`
  font-size: 1rem;
  margin-left: 0.3rem;
`;

const getGrowthColor = growth => {
  if (growth === 0) {
    return colors.blue;
  }

  return growth > 0 ? colors.green : colors.red;
};

const getIconName = growth => {
  if (growth === 0) {
    return "minus";
  }

  return growth > 0 ? "chevron-up" : "chevron-down";
};

export const KPICard = ({
  title,
  value,
  growth,
  redValue = false,
  greenValue = false
}) => (
  <Container>
    <Color textColor="#999999">{title}</Color>
    <Grid>
      <Cell middle>
        <Color
          textColor={
            (redValue && colors.red) || (greenValue && colors.green) || null
          }
          as={Value}
        >
          {new Intl.NumberFormat().format(value)}
        </Color>
      </Cell>
      <Cell center as={Bottom}>
        <FlexRow>
          <Color textColor={getGrowthColor(growth)}>
            {growth > 0 ? "+" : ""}
            {new Intl.NumberFormat("en-EN", {
              style: "percent",
              maximumFractionDigits: 2
            }).format(growth)}
            <Icon name={getIconName(growth)} as={TriangleIcon} />
          </Color>
        </FlexRow>
      </Cell>
    </Grid>
  </Container>
);

KPICard.propTypes = {
  /** Title for the box */
  title: PropTypes.string,
  /** Numeric value */
  value: PropTypes.number,
  /** Growth is converted from a fraction to percent, for example value 0.15 indicates 15% */
  growth: PropTypes.number,
  /** Defines whether the value should be red */
  redValue: PropTypes.bool,
  /** Defines whether the value should be green */
  greenValue: PropTypes.bool
};

KPICard.defaultProps = {
  title: "",
  value: 0,
  growth: 0,
  redValue: false,
  greenValue: false
};
