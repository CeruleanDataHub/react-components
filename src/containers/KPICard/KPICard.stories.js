import { withKnobs } from "@storybook/addon-knobs";
import React from "react";

import { Cell, Grid } from "../Grid/Grid";
import { KPICard } from "./KPICard";

export default {
  title: "KPICard",
  component: KPICard,
  decorators: [withKnobs]
};

export const KPICardStory = () => (
  <KPICard title="Value" value={0} growth={0} />
);

export const KPICardPositiveGrowthStory = () => (
  <KPICard title="Logins / week" value={3789} growth={0.145} />
);

export const KPICardNegativeGrowthStory = () => (
  <KPICard title="Logins / week" value={3789} growth={-0.145} />
);

export const KPICardZeroGrowthStory = () => (
  <KPICard title="Logins / week" value={3789} growth={0} />
);

export const KPICardGreenValueStory = () => (
  <KPICard title="Logins / week" value={3789} growth={0.145} greenValue />
);

export const KPICardRedValueStory = () => (
  <KPICard title="Logins / week" value={3789} growth={-0.145} redValue />
);

export const KPICardGreenValueZeroGrowthStory = () => (
  <KPICard title="Logins / week" value={3789} growth={0} greenValue />
);

export const KPICardRedValueZeroGrowthStory = () => (
  <KPICard title="Logins / week" value={3789} growth={0} redValue />
);

export const KPICardNoGrowthPercentageStory = () => (
  <KPICard
    title="Logins / week"
    value={3789}
    growth={0}
    showPercentage={false}
  />
);

export const KPICardGridStory = () => (
  <Grid>
    <Cell>
      <KPICard title="Logins / week" value={3789} growth={-0.145} />
    </Cell>
    <Cell>
      <KPICard title="Logins / month" value={17283} growth={0.87} />
    </Cell>
    <Cell>
      <KPICard
        title="Active users / week"
        value={1054}
        growth={-0.062}
        greenValue
      />
    </Cell>
    <Cell>
      <KPICard
        title="Active users / month"
        value={1870}
        growth={0.953}
        redValue
      />
    </Cell>
  </Grid>
);

export const KPICardEuro = () => (
  <KPICard title="Euro" value={3789} currency="EUR" dataFormat="currency" />
);

export const KPICardUSD = () => (
  <KPICard title="USD" value={3789} currency="USD" dataFormat="currency" />
);

export const KPICardEuroWithoutPercentage = () => (
  <KPICard
    title="Euro"
    value={3789}
    currency="EUR"
    dataFormat="currency"
    showPercentage={false}
  />
);

export const KPICardUSDWithoutPercentage = () => (
  <KPICard
    title="USD"
    value={3789}
    currency="USD"
    dataFormat="currency"
    showPercentage={false}
  />
);

export const KPICardUSDWithIcon = () => (
  <KPICard
    title="USD"
    icon="chef-hat"
    value={3789}
    currency="USD"
    dataFormat="currency"
  />
);

export const KPICardBackgroundColor = () => (
  <Grid>
    <Cell>
      <KPICard
        title="Logins / week"
        value={3789}
        backgroundColor="tomato"
        growth={-0.145}
      />
    </Cell>
    <Cell>
      <KPICard
        title="Logins / month"
        value={17283}
        backgroundColor="olive"
        growth={0.87}
      />
    </Cell>
    <Cell>
      <KPICard
        title="Active users / week"
        value={1054}
        growth={-0.062}
        backgroundColor="plum"
        greenValue
      />
    </Cell>
    <Cell>
      <KPICard
        title="Active users / month"
        value={1870}
        growth={0.953}
        backgroundColor="orange"
        redValue
      />
    </Cell>
  </Grid>
);

export const KPICardBorderRadius = () => (
  <KPICard
    title="Logins / week"
    value={3789}
    growth={0.145}
    backgroundColor="tomato"
    borderRadius="15px"
  />
);
