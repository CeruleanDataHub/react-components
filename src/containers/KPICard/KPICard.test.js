import React from "react";

import { mount } from "enzyme";

import { KPICard } from "./KPICard";

describe("KPICard", () => {
  let component;

  beforeEach(() => {
    component = mount(
      <KPICard
        title="Some title"
        value={42}
        growth={0}
        showPercentage={false}
        currency="EUR"
        dataFormat="currency"
        greenValue
        backgroundColor="tomato"
        borderRadius="15px"
      />
    );
  });

  it("renders", () => {
    expect(component).toMatchSnapshot();
  });

  it("given different currency, renders", () => {
    component = mount(<KPICard currency="USD" />);

    expect(component).toMatchSnapshot();
  });

  it("given different props, renders", () => {
    component = mount(
      <KPICard
        title="Some other title"
        growth={42}
        dataFormat="decimal"
        redValue
        showPercentage
      />
    );

    expect(component).toMatchSnapshot();
  });

  it("given negative growth, renders", () => {
    component = mount(<KPICard growth={-42} />);

    expect(component).toMatchSnapshot();
  });

  it("given large value, has formatted value", () => {
    component = mount(<KPICard value={420000000000} />);

    const formattedValue = component.find("div[data-cell-test]").text();

    expect(formattedValue).toEqual("420,000,000,000");
  });

  it("given large decimal, has formatted decimal", () => {
    component = mount(<KPICard growth={0.33333333} />);

    const formattedValue = component.find("span[data-flex-row-test]").text();

    expect(formattedValue).toEqual("+33.33%");
  });

  it("given icon, has icon", () => {
    component = mount(<KPICard icon="chef-hat" />);

    const icon = component.find("Icon[data-icon-test]");

    expect(icon).toExist();
  });
});
