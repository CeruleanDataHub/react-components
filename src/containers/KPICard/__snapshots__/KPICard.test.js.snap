// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KPICard should format large values 1`] = `
.c2 {
  height: 100%;
  min-width: 0;
  grid-column-end: span 1;
  grid-row-end: span 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-self: stretch;
}

.c5 {
  height: 100%;
  min-width: 0;
  grid-column-end: span 1;
  grid-row-end: span 1;
  text-align: center;
}

.c1 {
  display: grid;
  height: auto;
  grid-auto-flow: row;
  grid-auto-rows: minmax(20px,auto);
  grid-template-columns: repeat(2,1fr);
  grid-gap: 8px;
}

.c9 {
  font-size: 4rem;
  text-align: center;
}

.c0 {
  padding: 1.2em;
  font-family: Helvetica;
  font-size: 0.9rem;
}

.c7 {
  color: green;
}

.c3 {
  font-size: 2.2rem;
}

.c4 {
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c8 {
  font-size: 1rem;
  margin-left: 0.3rem;
}

<div
  className="c0"
>
  Title
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <span
        className="c3 "
      >
        10,000,000,000
      </span>
    </div>
    <span
      className="c4 c5"
    >
      <span
        className="c6"
      >
        <span
          className="c7"
        >
          +
          30%
          <div
            className="c8 c9 lni lni-chevron-up"
          />
        </span>
      </span>
    </span>
  </div>
</div>
`;

exports[`KPICard should render 1`] = `
.c2 {
  height: 100%;
  min-width: 0;
  grid-column-end: span 1;
  grid-row-end: span 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-self: stretch;
}

.c5 {
  height: 100%;
  min-width: 0;
  grid-column-end: span 1;
  grid-row-end: span 1;
  text-align: center;
}

.c1 {
  display: grid;
  height: auto;
  grid-auto-flow: row;
  grid-auto-rows: minmax(20px,auto);
  grid-template-columns: repeat(2,1fr);
  grid-gap: 8px;
}

.c9 {
  font-size: 4rem;
  text-align: center;
}

.c0 {
  padding: 1.2em;
  font-family: Helvetica;
  font-size: 0.9rem;
}

.c7 {
  color: green;
}

.c3 {
  font-size: 2.2rem;
}

.c4 {
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c8 {
  font-size: 1rem;
  margin-left: 0.3rem;
}

<div
  className="c0"
>
  Title
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <span
        className="c3 "
      >
        1
      </span>
    </div>
    <span
      className="c4 c5"
    >
      <span
        className="c6"
      >
        <span
          className="c7"
        >
          +
          30%
          <div
            className="c8 c9 lni lni-chevron-up"
          />
        </span>
      </span>
    </span>
  </div>
</div>
`;

exports[`KPICard should render green 1`] = `
.c2 {
  height: 100%;
  min-width: 0;
  grid-column-end: span 1;
  grid-row-end: span 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-self: stretch;
}

.c6 {
  height: 100%;
  min-width: 0;
  grid-column-end: span 1;
  grid-row-end: span 1;
  text-align: center;
}

.c1 {
  display: grid;
  height: auto;
  grid-auto-flow: row;
  grid-auto-rows: minmax(20px,auto);
  grid-template-columns: repeat(2,1fr);
  grid-gap: 8px;
}

.c9 {
  font-size: 4rem;
  text-align: center;
}

.c0 {
  padding: 1.2em;
  font-family: Helvetica;
  font-size: 0.9rem;
}

.c4 {
  color: green;
}

.c3 {
  font-size: 2.2rem;
}

.c5 {
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c7 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c8 {
  font-size: 1rem;
  margin-left: 0.3rem;
}

<div
  className="c0"
>
  Title
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <span
        className="c3 c4"
      >
        1
      </span>
    </div>
    <span
      className="c5 c6"
    >
      <span
        className="c7"
      >
        <span
          className="c4"
        >
          +
          30%
          <div
            className="c8 c9 lni lni-chevron-up"
          />
        </span>
      </span>
    </span>
  </div>
</div>
`;

exports[`KPICard should render negative growth 1`] = `
.c2 {
  height: 100%;
  min-width: 0;
  grid-column-end: span 1;
  grid-row-end: span 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-self: stretch;
}

.c5 {
  height: 100%;
  min-width: 0;
  grid-column-end: span 1;
  grid-row-end: span 1;
  text-align: center;
}

.c1 {
  display: grid;
  height: auto;
  grid-auto-flow: row;
  grid-auto-rows: minmax(20px,auto);
  grid-template-columns: repeat(2,1fr);
  grid-gap: 8px;
}

.c9 {
  font-size: 4rem;
  text-align: center;
}

.c0 {
  padding: 1.2em;
  font-family: Helvetica;
  font-size: 0.9rem;
}

.c7 {
  color: red;
}

.c3 {
  font-size: 2.2rem;
}

.c4 {
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c8 {
  font-size: 1rem;
  margin-left: 0.3rem;
}

<div
  className="c0"
>
  Title
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <span
        className="c3 "
      >
        1
      </span>
    </div>
    <span
      className="c4 c5"
    >
      <span
        className="c6"
      >
        <span
          className="c7"
        >
          
          -30%
          <div
            className="c8 c9 lni lni-chevron-down"
          />
        </span>
      </span>
    </span>
  </div>
</div>
`;

exports[`KPICard should render positive growth 1`] = `
.c2 {
  height: 100%;
  min-width: 0;
  grid-column-end: span 1;
  grid-row-end: span 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-self: stretch;
}

.c5 {
  height: 100%;
  min-width: 0;
  grid-column-end: span 1;
  grid-row-end: span 1;
  text-align: center;
}

.c1 {
  display: grid;
  height: auto;
  grid-auto-flow: row;
  grid-auto-rows: minmax(20px,auto);
  grid-template-columns: repeat(2,1fr);
  grid-gap: 8px;
}

.c9 {
  font-size: 4rem;
  text-align: center;
}

.c0 {
  padding: 1.2em;
  font-family: Helvetica;
  font-size: 0.9rem;
}

.c7 {
  color: green;
}

.c3 {
  font-size: 2.2rem;
}

.c4 {
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c8 {
  font-size: 1rem;
  margin-left: 0.3rem;
}

<div
  className="c0"
>
  Title
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <span
        className="c3 "
      >
        1
      </span>
    </div>
    <span
      className="c4 c5"
    >
      <span
        className="c6"
      >
        <span
          className="c7"
        >
          +
          30%
          <div
            className="c8 c9 lni lni-chevron-up"
          />
        </span>
      </span>
    </span>
  </div>
</div>
`;

exports[`KPICard should render red 1`] = `
.c2 {
  height: 100%;
  min-width: 0;
  grid-column-end: span 1;
  grid-row-end: span 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-self: stretch;
}

.c6 {
  height: 100%;
  min-width: 0;
  grid-column-end: span 1;
  grid-row-end: span 1;
  text-align: center;
}

.c1 {
  display: grid;
  height: auto;
  grid-auto-flow: row;
  grid-auto-rows: minmax(20px,auto);
  grid-template-columns: repeat(2,1fr);
  grid-gap: 8px;
}

.c10 {
  font-size: 4rem;
  text-align: center;
}

.c0 {
  padding: 1.2em;
  font-family: Helvetica;
  font-size: 0.9rem;
}

.c4 {
  color: red;
}

.c8 {
  color: green;
}

.c3 {
  font-size: 2.2rem;
}

.c5 {
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c7 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c9 {
  font-size: 1rem;
  margin-left: 0.3rem;
}

<div
  className="c0"
>
  Title
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <span
        className="c3 c4"
      >
        1
      </span>
    </div>
    <span
      className="c5 c6"
    >
      <span
        className="c7"
      >
        <span
          className="c8"
        >
          +
          30%
          <div
            className="c9 c10 lni lni-chevron-up"
          />
        </span>
      </span>
    </span>
  </div>
</div>
`;

exports[`KPICard should render two decimal points in growth 1`] = `
.c2 {
  height: 100%;
  min-width: 0;
  grid-column-end: span 1;
  grid-row-end: span 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-self: stretch;
}

.c5 {
  height: 100%;
  min-width: 0;
  grid-column-end: span 1;
  grid-row-end: span 1;
  text-align: center;
}

.c1 {
  display: grid;
  height: auto;
  grid-auto-flow: row;
  grid-auto-rows: minmax(20px,auto);
  grid-template-columns: repeat(2,1fr);
  grid-gap: 8px;
}

.c9 {
  font-size: 4rem;
  text-align: center;
}

.c0 {
  padding: 1.2em;
  font-family: Helvetica;
  font-size: 0.9rem;
}

.c7 {
  color: green;
}

.c3 {
  font-size: 2.2rem;
}

.c4 {
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c8 {
  font-size: 1rem;
  margin-left: 0.3rem;
}

<div
  className="c0"
>
  Title
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <span
        className="c3 "
      >
        1
      </span>
    </div>
    <span
      className="c4 c5"
    >
      <span
        className="c6"
      >
        <span
          className="c7"
        >
          +
          33.33%
          <div
            className="c8 c9 lni lni-chevron-up"
          />
        </span>
      </span>
    </span>
  </div>
</div>
`;

exports[`KPICard should render zero growth 1`] = `
.c2 {
  height: 100%;
  min-width: 0;
  grid-column-end: span 1;
  grid-row-end: span 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-self: stretch;
}

.c5 {
  height: 100%;
  min-width: 0;
  grid-column-end: span 1;
  grid-row-end: span 1;
  text-align: center;
}

.c1 {
  display: grid;
  height: auto;
  grid-auto-flow: row;
  grid-auto-rows: minmax(20px,auto);
  grid-template-columns: repeat(2,1fr);
  grid-gap: 8px;
}

.c9 {
  font-size: 4rem;
  text-align: center;
}

.c0 {
  padding: 1.2em;
  font-family: Helvetica;
  font-size: 0.9rem;
}

.c7 {
  color: #1ea7fd;
}

.c3 {
  font-size: 2.2rem;
}

.c4 {
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c8 {
  font-size: 1rem;
  margin-left: 0.3rem;
}

<div
  className="c0"
>
  Title
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <span
        className="c3 "
      >
        1
      </span>
    </div>
    <span
      className="c4 c5"
    >
      <span
        className="c6"
      >
        <span
          className="c7"
        >
          
          0%
          <div
            className="c8 c9 lni lni-minus"
          />
        </span>
      </span>
    </span>
  </div>
</div>
`;
